cmd/server-main.go

```diff
@@ -485,62 +377,115 @@ func serverMain(c *cli.Context) {
-	// as command line args are local to this server.
-	if !isAnyEndpointLocal(endpoints) {
-		fatalIf(errInvalidArgument, "None of the disks passed as command line args are local to this server.")
-	}
+	endpoints, err := parseStorageEndpoints(ctx.Args())
+	fatalIf(err, "Unable to parse storage endpoints %s", ctx.Args())
 
 	// Sort endpoints for consistent ordering across multiple
 	// nodes in a distributed setup. This is to avoid format.json
 	// corruption if the disks aren't supplied in the same order
 	// on all nodes.
 	sort.Sort(byHostPath(endpoints))
 
-	// Configure server.
-	srvConfig := serverCmdConfig{
-		serverAddr: serverAddr,
-		endpoints:  endpoints,
+	checkServerSyntax(endpoints, ctx.Args())
+
+	// Should exit gracefully if none of the endpoints passed
+	// as command line args are local to this server.
+	if !isAnyEndpointLocal(endpoints) {
+		fatalIf(errInvalidArgument, "None of the disks passed as command line args are local to this server.")
 	}
 
 	// Check if endpoints are part of distributed setup.
 	globalIsDistXL = isDistributedSetup(endpoints)
 
-	// Set nodes for dsync for distributed setup.
-	if globalIsDistXL {
-		fatalIf(initDsyncNodes(endpoints), "Unable to initialize distributed locking clients")
-	}
-
 	// Set globalIsXL if erasure code backend is about to be
 	// initialized for the given endpoints.
 	if len(endpoints) > 1 {
 		globalIsXL = true
 	}
 
+	// Set endpoints of []*url.URL type to globalEndpoints.
+	globalEndpoints = endpoints
+}
+
+func handleEnvVars() {
+	// Start profiler if env is set.
+	if profiler := os.Getenv("_MINIO_PROFILER"); profiler != "" {
+		globalProfiler = startProfiler(profiler)
+	}
+
+	// Check if object cache is disabled.
+	globalObjCacheDisabled = strings.EqualFold(os.Getenv("_MINIO_CACHE"), "off")
+
+	accessKey := os.Getenv("MINIO_ACCESS_KEY")
+	secretKey := os.Getenv("MINIO_SECRET_KEY")
+	if accessKey != "" && secretKey != "" {
+		cred, err := createCredential(accessKey, secretKey)
+		fatalIf(err, "Invalid access/secret Key set in environment.")
+
+		// credential Envs are set globally.
+		globalIsEnvCreds = true
+		globalActiveCred = cred
+	}
+
+	if browser := os.Getenv("MINIO_BROWSER"); browser != "" {
+		browserFlag, err := ParseBrowserFlag(browser)
+		if err != nil {
+			fatalIf(errors.New("invalid value"), "Unknown value ‘%s’ in MINIO_BROWSER environment variable.", browser)
+		}
+
+		// browser Envs are set globally, this does not represent
+		// if browser is turned off or on.
+		globalIsEnvBrowser = true
+		globalIsBrowserEnabled = bool(browserFlag)
+	}
+}
+
+// serverMain handler called for 'minio server' command.
+func serverMain(c *cli.Context) {
+	if !c.Args().Present() || c.Args().First() == "help" {
+		cli.ShowCommandHelpAndExit(c, "server", 1)
+	}
+
+	// Get quiet flag from command line argument.
+	quietFlag := c.Bool("quiet") || c.GlobalBool("quiet")
+	if quietFlag {
+		log.EnableQuiet()
+	}
+
+	handleCmdArgs(c)
+	handleEnvVars()
```

Comment:
prefix this with `serverHandleCmdArgs()` so that avoiding namespace conflict in future for other commands.